path.go

```diff
@@ -3,22 +3,31 @@ package txtdirect
 import (
 	"context"
 	"fmt"
+	"log"
 	"net"
 	"regexp"
 	"sort"
 	"strconv"
 	"strings"
+	"time"
 )
 
 var PathRegex = regexp.MustCompile("\\/([A-Za-z0-9-._~!$'()*+,;=:@]+)")
 var FromRegex = regexp.MustCompile("\\/\\$(\\d+)")
 
 func zoneFromPath(host string, path string, rec record) (string, int, error) {
 	pathSubmatchs := PathRegex.FindAllStringSubmatch(path, -1)
+	if rec.Re != "" {
+		CustomRegex := regexp.MustCompile(rec.Re)
```

Comment:
We might want to use `regex.Compile` here and check for the error, otherwise a badly crafted regex could crash the server